src/components/KTable/KTable.vue

```diff
@@ -835,6 +856,29 @@ const getTestIdString = (message: string) => {
   return msg
 }
 
+watch(fetcherData, () => {
+  if (fetcherData.value?.length && !data.value.length) {
+    data.value = fetcherData.value
+  }
+}, { deep: true, immediate: true })
+
+// we want to tie loader to 'pending' since 'validating' is triggered even when pulling from cache, which should result in no loader
+// however, if this is a manual revalidation (triggered by page change, query, etc), display loader when validating
+watch(state, () => {
+  switch (state.value) {
+    case swrvState.PENDING:
+      isTableLoading.value = true
+      break
+    case swrvState.VALIDATING_HAS_DATA:
+      isRevalidating.value ? isTableLoading.value = true : isTableLoading.value = false
```

Comment:
suggestion: can we simplify this?
```suggestion
      isTableLoading.value = isRevalidating.value
```